engine/gamesys/src/gamesys/scripts/script_window.cpp

```diff
@@ -291,22 +291,99 @@ static int GetSize(lua_State* L)
  */
 static int GetMouseLock(lua_State* L)
 {
-    int top = lua_gettop(L);
+    DM_LUA_STACK_CHECK(L, 1);
+
     bool cursor_visible = dmHID::GetCursorVisible(g_Window.m_HidContext);
     // If cursor is visible, it is not locked
     lua_pushboolean(L, !cursor_visible);
 
-    assert(top + 1 == lua_gettop(L));
     return 1;
 }
 
+/*# get the display scale
+ *
+ * This returns the content scale of the current display.
+ *
+ * @name window.get_display_scale
+ * @return scale [type:number] The display scale
+ */
+static int GetScale(lua_State* L)
+{
+    DM_LUA_STACK_CHECK(L, 1);
+
+    float scale = dmPlatform::GetDisplayScaleFactor(g_Window.m_Window);
+    lua_pushnumber(L, scale);
+
+    return 1;
+}
+
+/*# set the title of the window
+ *
+ * Sets the window title.
+ *
+ * @name window.set_title
+ * @param title [type:string] The title, encoded as UTF-8
+ */
+static int SetTitle(lua_State* L)
+{
+    DM_LUA_STACK_CHECK(L, 0);
+
+    const char* title = luaL_checkstring(L, 1);
+    dmPlatform::SetWindowTitle(g_Window.m_Window, title);
+
+    return 0;
+}
+
+/*# set the size of the window
+ *
+ * Sets the window size.
+ *
+ * @name window.set_size
+ * @param width [type:number] Width of window
+ * @param height [type:number] Height of window
+ */
+static int SetSize(lua_State* L)
+{
+    DM_LUA_STACK_CHECK(L, 0);
+
+    int width = luaL_checkinteger(L, 1);
+    int height = luaL_checkinteger(L, 2);
+    dmPlatform::SetWindowSize(g_Window.m_Window, width, height);
+
+    return 0;
+}
+
+
+/*# set the position of the window
+ *
+ * Sets the window position.
+ *
+ * @name window.set_position
+ * @param x [type:number] Horizontal position of window
+ * @param y [type:number] Vertical position of window
+ */
+static int SetPosition(lua_State* L)
+{
+    DM_LUA_STACK_CHECK(L, 0);
+
+    int x = luaL_checkinteger(L, 1);
+    int y = luaL_checkinteger(L, 2);
+    dmPlatform::SetWindowPosition(g_Window.m_Window, x, y);
+
+    return 0;
+}
+
 static const luaL_reg Module_methods[] =
 {
     {"set_listener",   SetListener},
     {"set_dim_mode",   SetDimMode},
     {"set_mouse_lock", SetMouseLock},
+    {"set_title",      SetTitle},
     {"get_dim_mode",   GetDimMode},
     {"get_size",       GetSize},
+    {"set_size",       SetSize},
+    {"set_position",   SetPosition},
+    {"get_scale",      GetScale},
```

Comment:
Mismatch between name "get_scale" vs "get_display_scale" in names/comments (also in PR description)